src/indexer/migrations/20241009154948_create_tables.sql

```diff
@@ -1,30 +1,30 @@
 -- Add migration script here
 CREATE TABLE blocks (
-    hash BYTEA PRIMARY KEY,         -- Corresponds to BlockHash (inner Vec<u8>)
-    parent_hash BYTEA NOT NULL,     -- Parent block hash (BlockHash)
+    hash TEXT PRIMARY KEY,          -- Corresponds to BlockHash 
+    parent_hash TEXT NOT NULL,      -- Parent block hash (BlockHash)
     height BIGINT NOT NULL,         -- Corresponds to BlockHeight (u64)
     timestamp TIMESTAMP NOT NULL,   -- UNIX timestamp (u64)
     UNIQUE (height),                -- Ensure each block height is unique
-    CHECK (octet_length(hash) = 32),-- Ensure the hash is exactly 32 bytes
+    CHECK (length(hash) = 64),      -- Ensure the hash is exactly 64
     CHECK (height >= 0)             -- Ensure the height is positive
 );
 
 CREATE TYPE transaction_type AS ENUM ('blob_transaction', 'proof_transaction', 'register_contract_transaction');
 CREATE TYPE transaction_status AS ENUM ('success', 'failure', 'sequenced');
 
 CREATE TABLE transactions (
-    tx_hash BYTEA PRIMARY KEY,
-    block_hash BYTEA NOT NULL REFERENCES blocks(hash) ON DELETE CASCADE,
+    tx_hash TEXT PRIMARY KEY,
```

Comment:
Au lieu de Text + check, il y a VARCHAR(64)

Reply:
ou même CHARACTER(64)

mais c'est exactements [identiques en matière de perfs](https://www.postgresql.org/docs/current/datatype-character.html)

> There is no performance difference among these three types, apart from increased storage space when using the blank-padded type, and a few extra CPU cycles to check the length when storing into a length-constrained column. While character(n) has performance advantages in some other database systems, there is no such advantage in PostgreSQL; in fact character(n) is usually the slowest of the three because of its additional storage costs. In most situations text or character varying should be used instead.